### PR TITLE
Add GCE Ingress in addition to NGINX Ingress

### PR DESCRIPTION
**Pull request by lavjain**
GCE Ingress has support for IAP as well as Google managed certificates

### CODE REVIEW

**Review by jonchenn — CHANGES_REQUESTED** [review]
One suggestion on the module folder name:
- ingress_gce
- ingress_nginx
So these two folders will be next to each other in the folder order.

**Comment by lavjain** [comment]
Rebased with main

File: terraform/stages/gke/main.tf [hunk 1]
```diff
@@ -31,23 +31,22 @@ module "gke" {
...
   # See latest stable version at https://cloud.google.com/kubernetes-engine/docs/release-notes-stable
-  kubernetes_version = "1.23.14-gke.1800"
+  kubernetes_version = "1.23.16-gke.1400"
 }
 
 module "ingress" {
   depends_on = [module.gke]
 
-  source            = "../../modules/ingress"
+  source            = "../../modules/ingress_gce"
```
Comment (jonchenn) [inline terraform/stages/gke/main.tf]: Can you revert this back to using the original nginx-ingress until we have tested it throughouly?

File: build_tools/build_template.sh
Comment (jonchenn) [inline build_tools/build_template.sh:1]: Not sure why it's showing empty file? It should be here: https://github.com/GoogleCloudPlatform/solutions-template/blob/main/build_tools/build_template.sh

File: terraform/stages/gke/providers.tf [hunk 2]
```diff
@@ -28,7 +28,7 @@ terraform {
     }
     helm = {
       source  = "hashicorp/helm"
-      version = ">= 2.5.1"
+      version = ">= 2.7.0"
```
Comment (jonchenn) [inline terraform/stages/gke/providers.tf:31]: We need to test this.

**Review by jonchenn — CHANGES_REQUESTED** [review]
Left a few comments, can you revert the default usage of ingress to the "ingress_nginx"?

File: build_tools/build_template.sh
Comment (lavjain) [reply build_tools/build_template.sh:1]: I changed the permission to 755 from 644

File: terraform/stages/gke/providers.tf (hunk 2 above)
Comment (lavjain) [reply terraform/stages/gke/providers.tf:31]: The helm version is 2.7.0 in terraform/modules/ingress/main.tf (we should maintain consistent versions)

I have been testing it by creating a solutions template installation from scratch

Comment (jonchenn) [reply terraform/stages/gke/providers.tf:31]: Sg, I'll run e2e-test after the update of the PR.

File: terraform/stages/gke/main.tf (hunk 1 above)
Comment (lavjain) [reply terraform/stages/gke/main.tf]: Done

**Review by jonchenn — COMMENTED** [review]
LGTM, but please run e2e-test with this PR.

**Review by jonchenn — APPROVED** [review]
LGTM

**Pull request merged**